movies-app/src/components/GenreSelect/GenreSelect.jsx

```diff
@@ -1,30 +1,54 @@
-import React, { useState, useEffect} from "react";
-import './GenreSelect.css';
+import React from "react";
+import "./GenreSelect.css";
+import { useSearchParams } from "react-router-dom";
 
 const GenreSelect = (props) => {
 
-    const onSelectHandler = (event) => {
-      const genre = event.target.innerHTML;
-      props.onSelect(genre) 
+  const [searchParams, setSearchParams] = useSearchParams({});
+
+  const activeGenre = searchParams.get("activeGenre") ? searchParams.get("activeGenre") : "All";
+  const sortBy = searchParams.get("sortBy");
+
+  const onSelectHandler = (event) => {
+
+    const selectedGenre = event.target.innerHTML;
```

Comment:
Security szempontból szerencsésebb
```suggestion
    const selectedGenre = event.target.innerText;
```